subsys/sal/sid_pal/src/sid_ble_adapter.c

```diff
@@ -213,14 +214,35 @@ static sid_error_t ble_adapter_set_tx_pwr(int8_t tx_power)
 	return SID_ERROR_NONE;
 }
 
+static int create_ble_id(void)
+{
+	int ret;
+	size_t count = 0;
+
+	/* Check if Bluetooth identites weren't already created. */
+	bt_id_get(NULL, &count);
+	if (count > BT_ID_SIDEWALK) {
+		return 0;
+	}
+
+	do {
+		ret = bt_id_create(NULL, NULL);
+		if (ret < 0) {
+			return ret;
+		}
+	} while (ret != BT_ID_SIDEWALK);
```

Comment:
risk of inifinite loop?

Why isn't the id saved in some static struct and compared in callback? I see something like this is done in nrodic_dfu and some sources in ncs.